Dev

LunexCoding opened the pull request.

LunexCoding committed:
v0.0.1 Parser setup
A way to download a dataset to a local computer.

`Calendar` is used to get the date of the partial download of the
 datasets.

Creating a directory tree with `FileSystem`.

In the plans:
     * Alteration of `Calendar`;
     * Refinement of the parser by checking the availability and
 relevance of the data set.

LunexCoding committed:
Fix timing errors, improve parser logic

`Parser`:
Previously, an incomplete date for uploading a dataset to the site
 was supplied. Now the date of uploading the data set to the site
 is supplied in full.

`Calendar`:
Now all the time will be converted to `GMT` format for convenience,
 since the time supplied from the site is specified in this format.

`FileSystem`:
Added the ability to get the last modification date of a file.

`.gitignore`:
Typo fixed: `.__pycache__/` -> `__pycache__/`

Resolves: #1, #2, #3, #4

oleksandr-grabovyi commented on Calendar.py:
```diff
@@ -0,0 +1,46 @@
+
+
+class MONTHS(enum.IntEnum):
+    Январь = 1
```
нельзя использовать кириллицу в имени переменной

oleksandr-grabovyi commented on Calendar.py:
```diff
@@ -0,0 +1,46 @@
+        self._serverDate = None
+
+    def timeToGMT(self, data):
+        if isinstance(data, float):
```
плохая идея. Если тебе нужно получать из float и из строки, то лучше сделай 2 отдельных метода. getGMTFromFloat, getGMTFromStr

oleksandr-grabovyi commented on fileSystem.py:
```diff
@@ -0,0 +1,44 @@
+
+    def checkFileExists(self, filename):
+        if os.path.exists(filename):
+            return self._getFileLastModifiedDate(filename)
```
Функция _getFileLastModificationDate разве бульку возвращает? Тут вообще возврат выглядит типа:
return os.path.exists(filename)

oleksandr-grabovyi commented on fileSystem.py:
```diff
@@ -0,0 +1,44 @@
+
+
+g_fileSystem = FileSystem('assets')
+g_fileSystem._createTree([
```
как-то не хорошо так прям инициализировать все. Если это требуется в файловой системе всегда. Ну например - то лучше уж в конструктор перенести. А если ты не хочешь именно в конструктор - то в каком-то мейне инициализируй пути. Потому как тут сначала создается глобальная переменная, потом сама себя заполняет. Ну такое шото. + метод защищенный типа. Не оч хорошо вызывать извне

oleksandr-grabovyi commented on parser.py:
```diff
@@ -0,0 +1,63 @@
+
+        g_fileSystem.chdir(g_fileSystem.root)
+
+    def run(self):
```
run обычно предполагает какой-то процесс. А тут единичный запуск функции. start или init больше подойдет

LunexCoding committed:
Fix some logical errors

LunexCoding committed:
Add logging, replace os.path with pathlib

The old module [Logger](https://github.com/LunexCoding/Logger.git)
 has been reworked. It is planned to finalize it by adding
 customization options.

To simplify working with paths, `os.path` has been replaced with
 `pathlib`.

Resolves: #4, #6

oleksandr-grabovyi commented on Calendar.py:
```diff
@@ -0,0 +1,29 @@
+    def getGMTFromFloat(self, data):
+        dtUTC = datetime.utcfromtimestamp(data)
+        timeStringFormatFromdDT = datetime.strftime(dtUTC, "%a, %d %b %Y %H:%M:%S GMT")
+        return datetime.strptime(timeStringFormatFromdDT, "%a, %d %b %Y %H:%M:%S GMT")
```
вот тут по идеи можно вызвать getGMTFromStr. Ты ведь стрингу уже получил как бы. Но это не обязательно. Логика в принципе может и отличатся для флоата и строк.

oleksandr-grabovyi commented on parser.py:
```diff
@@ -0,0 +1,70 @@
+        g_fileSystem.chdir('sets')
+
+        for dataSet in zip(self._dataSetLinks, self._dataSetRegions):
+
```
наверное лишняя пустая строка ) как-то не красиво)

oleksandr-grabovyi commented on parser.py:
```diff
@@ -0,0 +1,70 @@
+            try:
+                serverDate = g_calendar.getGMTFromStr(responce.headers['Last-Modified'])
+                localDate = g_fileSystem.getFileLastModifiedGMTDate(filename)
+
```
не плоди много лишних строк)

oleksandr-grabovyi commented on parser.py:
```diff
@@ -0,0 +1,70 @@
+        self._generateSoup()
+        self._dataSetLinks = self._getDataSetLinks()
+        self._dataSetRegions = self._getDataSetRegions()
+        self._downloadDataSets()
```
я так понимаю вот тут надо и написать будет что-то типа if g_settings.isAutoDownload:

LunexCoding committed:
Add base config file

If there is no configuration file, it will be created and filled with
 default values.

Resolves: #7

LunexCoding committed:
Add autoload switch

`parser` -> Autoload:
Default is `False`, can be changed in config to `True`.

`settings`:
The creation of the config has been redesigned, you need to call the
 `load` method, and accessing variables.

`fileSystem`:
Now the root will be generated in the tree, and not separately from
 the tree, that is, you will need to specify a relative path:
`Path('sets')` -> `Path('assets', 'sets')`
This helped avoid the following bug:
the config was created before the log.

LunexCoding committed:
Remove root initialization

oleksandr-grabovyi commented on settings.py:
```diff
@@ -0,0 +1,47 @@
+            self._config.write(configFile, space_around_delimiters=True)
+
+
+g_config = Config('config.ini')
```
не вижу в ревью этот файл. Ты забыл залить наверное

oleksandr-grabovyi commented on parser.py:
```diff
@@ -0,0 +1,68 @@
+        g_fileSystem.chdir('sets')
+        for dataSet in zip(self._dataSetLinks, self._dataSetRegions):
+            url, region = dataSet
+            filename = region + '.json'
```
эм. Ну какой-то хардкод получается. Во первых: всегда json должен быть?
Во вторых ты даже не проверяешь, есть ли уже окончание файла. Вдруг он уже myfile.json называется.
Я бы написал какой-то метод типа updateFileExtension(filename, extension) и в нем проверял. Если в конце файла уже стоит такое - то ниче не делал, а если нет - то возвращал конкатенацию строк. Ну что-то такое. +даже когда будешь вызывать метод, не надо писать аля updateFileExtension(region, '.json') надо вынести в какую-то отдельную константу аля: DATA_SETS_EXTENSION = '.json' и ее использовать. Потом захочешь когда-то перейти на другой тип файлов - заменишь в 1 месте

oleksandr-grabovyi commented on main.py:
```diff
@@ -0,0 +1,26 @@
+from parser import g_parser
+from Logger import logger
+from fileSystem import g_fileSystem
+from pathlib import Path
```
обычно рекомендуют стандартные библиотеки писать в начале файла, а твои импорты уже после них

oleksandr-grabovyi commented on fileSystem.py:
```diff
@@ -0,0 +1,42 @@
+        return self._root
+
+
+g_fileSystem = FileSystem('assets')
```
слуш. Ну это ты тогда не g_fileSystem создал. Если у тебя файловая системая принимает какую-то папку и т.д. То ты можешь создать получается уже несколько файловых систем в зависиомсти от папки. Следовательно и название должно это как-то отражать g_assetsFileSystem и т.д.
